### PR TITLE
HAWQ-177. finalize resource queue ddl error messages

### PR DESCRIPTION
**Pull request by jiny2**
This fix is to finalize all resource queue ddl CREATE/ALTER/DROP RESOURCE QUEUE error messages. Then they will be verified by test cases.
For the specifications of the ddl and its error messages, please check 
HAWQ 2.0 Resource Management Design (+FTS), section 4.4

### CODE REVIEW

**Comment by linwen** [comment]
looks good.

File: src/backend/resourcemanager/resqueuemanager.c [hunk 1]
```diff
@@ -1065,16 +1070,27 @@ int checkAndCompleteNewResourceQueueAttributes(DynResourceQueue  queue,
...
 			return res;
 		}
+
+		/* The parent queue cannot have roles assigned. */
+		if ( hasUserAssignedToQueue(parenttrack->QueueInfo) )
+		{
+			res = RESQUEMGR_WRONG_ATTR;
+			snprintf(errorbuf, errorbufsize,
+					 "%s has at least one role assigned",
+					 parenttrack->QueueInfo->Name);
```
Comment (yaoj2) [inline src/backend/resourcemanager/resqueuemanager.c]: What about 
"parent quene %s should not have roles assigned"

File: src/backend/resourcemanager/resqueuemanager.c [hunk 2]
```diff
@@ -1511,9 +1521,7 @@ int createQueueAndTrack( DynResourceQueue		queue,
 			{
 				res = RESQUEMGR_WRONG_PARENT_QUEUE;
 				snprintf( errorbuf, errorbufsize,
-						  ERRORPOS_FORMAT
-						  "The parent queue of can not be pg_default.",
-						  ERRREPORTPOS);
+						  "the parent queue can not be pg_default");
```
Comment (yaoj2) [inline src/backend/resourcemanager/resqueuemanager.c]: it should be cannot here

**Comment by yaoj2** [comment]
I suggest that the error message we should directly tell user what should be. 
For example: 
ERROR: the values of memory_limit_cluster and core_limit_cluster are not identical
->
ERROR: the values of memory_limit_cluster and core_limit_cluster should be identical

ERROR: active_statements is less than 1
->
ERROR: active_statements should be greater than 1

nvseg_upper_limit_perseg is less than nvseg_lower_limit_perseg
->
nvseg_upper_limit_perseg should be greater than nvseg_lower_limit_perseg
2. Is it any reason we remove 64MB  for vseg_resource_quota?

3. drop a queue having at least one child queue.
What about change the message to "resource queue %s has at least one chile queue."

File: src/backend/resourcemanager/resqueuemanager.c (hunk 1 above)
Comment (jiny2) [inline src/backend/resourcemanager/resqueuemanager.c]: I prefer using the message which explain the error reason.

File: src/backend/resourcemanager/resqueuemanager.c (hunk 2 above)
Comment (jiny2) [inline src/backend/resourcemanager/resqueuemanager.c]: Fixed Thanks.

**Pull request closed without merging**